lib/perl/Genome/Model/Tools/EpitopePrediction/NewPipeline.pm

```diff
@@ -0,0 +1,161 @@
+package Genome::Model::Tools::EpitopePrediction::NewPipeline;
+
+use strict;
+use warnings;
+use Genome;
+use File::Basename qw(dirname);
+use YAML;
+
+class Genome::Model::Tools::EpitopePrediction::NewPipeline {
+    is => 'Command::V2',
+    has_input => [
+        output_directory => {
+            is => 'Text',
+            doc => 'the directory where you want results stored',
+        },
+        build => {
+            is => "Genome::Model::Build::SomaticVariation",
+        },
+        alleles => {
+            is => 'Text',
+            doc => 'A list of allele names to be used for epitope prediction with NetMHC',
+            is_many => 1,
+        },
+        epitope_length => {
+            is => 'Text',
+            doc => 'Length of subpeptides to predict with NetMHC',
+        },
+        netmhc_version => {
+            is => 'Text',
+            doc => 'The NetMHC version to use',
+            valid_values => ['3.0','3.4'],
+            default_value => '3.4',
+        },
+        output_filter => {
+            is => 'Text',
+            doc =>
+                'Type of epitopes to report in the final output - select \'top\' to report the top epitopes in terms of fold changes,  \'all\' to report all predictions ',
+            valid_values => ['top', 'all'],
+        },
+    ],
+};
+
+sub process_class {
+    return "Genome::Model::Tools::EpitopePrediction::Process";
+}
+
+sub execute {
+    my $self = shift;
+
+    $self->validate_inputs;
+
+    my $p = $self->process_class->create(
+        $self->process_params,
+    );
+
+    $self->status_message("Constructing workflow from inputs. (this may take a while...)");
+    $p->run(
+        workflow_xml => $self->dag->get_xml,
+        workflow_inputs => $self->workflow_inputs($p->id),
+    );
+
+    return $p;
+}
+
+sub validate_inputs {
+    my $self = shift;
+
+    if (-d $self->output_directory) {
+        $self->fatal_message("Output directory %s already exists", $self->output_directory);
+    }
+
+    Genome::Sys::make_path($self->output_directory);
```

Comment:
This should use `Genome::Sys->create_directory` to make sure permissions are set appropriately.